plugins/schedule/base/lib/iCal/kSchedulingICalEvent.php

```diff
@@ -195,26 +198,51 @@ public static function fromObject(KalturaScheduleEvent $event)
 		if ($event->referenceId)
 			$object->setField('uid', $event->referenceId);
 
+		if ($event->recurrence && $event->recurrence->timeZone)
+		{
+			$timeZones = DateTimeZone::listIdentifiers();
+
+			if (in_array($event->recurrence->timeZone, $timeZones))
+			{
+				$object->timeZoneId = $event->recurrence->timeZone;
+			}
+		}
+
 		foreach (self::$stringFields as $string)
 		{
 			if ($event->$string)
 			{
 				if ($string == 'duration')
 				{
+					if (!is_null($event->endDate))

```

Comment:
apparently the ics can have end date and duration together. it should be one or the other. Actually i think it can technically, but this was one of the remarks I got from Arek/Extron